types/cosmos.go

```diff
@@ -155,52 +155,34 @@ func (tx Tx) Successful() bool {
 
 // Message represents the data of a single message
 type Message struct {
-	TxHash    string
-	Index     int
-	Type      string
-	Value     string
-	Addresses []string
-	Height    int64
+	TxHash string
+	Index  int
+	// Message type url
+	Type string
+	// Value that can be used to create a Any object to
+	// unmarshal the message
+	Value  []byte
+	Height int64
 }
 
 // NewMessage allows to build a new Message instance
-func NewMessage(txHash string, index int, msgType string, value string, addresses []string, height int64) *Message {
+func NewMessage(txHash string, index int, msgType string, value []byte, height int64) *Message {
 	return &Message{
-		TxHash:    txHash,
-		Index:     index,
-		Type:      msgType,
-		Value:     value,
-		Addresses: addresses,
```

Comment:
Since we don't decode the message, I can't determine which accounts are involved in a message.